src/coreclr/src/classlibnative/bcltype/CMakeLists.txt

```diff
@@ -10,9 +10,11 @@ set(BCLTYPE_SOURCES
     variant.cpp
 )
 
-add_library_clr(bcltype
+add_library_clr(bcltype_obj
     OBJECT
     ${BCLTYPE_SOURCES}
 )
 
-add_dependencies(bcltype eventing_headers)
+add_dependencies(bcltype_obj eventing_headers)
+add_library(bcltype INTERFACE)
```

Comment:
An idea for a future cleanup: This add_library / target_sources pair pattern is used at 24 places. It may be nice to clean this up in a follow up change by creating a cmake function for this pattern. Or even encapsulate the add_library_clr in that function, getting something along these lines:
```cmake
add_intermediate_library_clr(bcltype ${BCLTYPE_SOURCES})
add_intermediate_library_dependencies(bcltype eventing_headers)
```

Reply:
Yes, we should clean this up in the future by making some additional functions.